GHCN.py

```diff
@@ -0,0 +1,1044 @@
+    if varmap is None: varmap = ghcn_varmap
+    elif not isinstance(varmap, dict): raise TypeError
+    # utils
+    encoding = encoding or variables.values()[0].encoding 
+    if atts is None: atts = dict(name=datatype, title=title) # default name
+    elif not isinstance(atts,dict): raise TypeError # resulting dataset attributes
+    if not isinstance(encoding,basestring): raise TypeError
+    folder = folder #or '{:s}/{:s}_{:s}/'.format(root_folder,interval,datatype) # default folder scheme 
+    if not isinstance(folder,basestring): raise TypeError
+    # save arguments
+    self.folder = folder
+    self.stationfile = stationfile
+    self.encoding = encoding
+    self.interval = interval
+    self.datatype = datatype
+    self.title = title
+    self.variables = variables
+    self.extremes = extremes
+    self.varmap = varmap
+    self.ravmap = dict((value,key) for key,value in varmap.iteritems() if value is not None) # reverse var map
+    self.atts = atts
+    self.header_format = header_format
+    self.station_format = station_format
+    self.constraints = constraints
+    ## initialize station objects from file
+    # open and parse station file
+    stationfile = '{:s}/{:s}'.format(folder,stationfile)
+    f = codecs.open(stationfile, 'r', encoding=encoding)
+    # initialize station list
+    self.stationlists = {varname:[] for varname in variables.iterkeys()} # a separate list for each variable
+    z = 0 # row counter 
+    ns = 0 # station counter
+    # loop over lines (each defiens a station)
+    for line in f:
+      z += 1 # increment counter
+      collist = line.split()
+      if len(collist) > 0: # skip empty lines
+        stdef = dict() # station specific arguments to instantiate station object
+        # loop over column titles
+        zz = 0 # column counter
+        for key,fct in station_format[0:5]: # loop over columns
+          if key is None: # None means skip this column
+            if zz == 0: # first column
+              if z != fct(collist[zz]): raise ParseError, "Station number is not consistent with line count:\n {:s}".format(line)
+          else:
+            stdef[key] = fct(collist[zz]) # convert value and assign to argument
+          zz += 1 # increment column
+        assert zz <= len(collist) # not done yet
+        for key,fct in station_format[-4:]:
+           stdef[key]=fct('0')
+        if constraints is None: ladd = True
+        else:
+          ladd = True
+          for key,val in constraints.iteritems():
+            if stdef[key] not in val: ladd = False
+        # instantiate station objects for each variable and append to lists
+        if ladd:
+          ns += 1
+          # loop over variable definitions
+          for varname,vardef in variables.iteritems():
+            filename = '{0:s}/ghcnd_all/{1:s}.dly'.format(folder, format(stdef['id']))
+            kwargs = dict() # combine station and variable attributes

```

Comment:
I'm assuming you change ghcnd_all when you want to run a test with 4 or 10 stations. It would be OK to introduce a debug-mode, so that you just have to pass an argument, e.g. ldebug=True to run a test with less stations.